### ISSUE TITLE
ReadWriteSplittingPlugin should (optionally) replace cluster url with writer node url to enable connection pooling for writer connections

### ISSUE THREAD

**Issue by ivankhoosty** [aws/aws-advanced-jdbc-wrapper#999]
### Describe the feature

Most Spring/SpringBoot apps use (external in driver terminology) connection pooling, probably default Hikari. 
ReadWriteSplittingPlugin documentation clearly states that "connections to cluster URL will not be pooled".  This means that when external pool is disabled, and driver internal pool configuration is enabled, (writer) connections to cluster URL are no longer pooled, and only reader node connections are pooled. This results in unacceptable performance degradation for connections to writer nodes via cluster URL.

If ReadWriteSplittingPlugin could (optionally, so no impact change) replace cluster url with writer node url, internal connection pool for writer node connections would offer the same performance as external pool.

### Use Case

I want to use ReadWriteSplittingPlugin to take advantage of reader nodes, but I cannot afford to lose connection pooling for writer node connections in a SpringBoot application.

ReadWriteSplittingPlugin documentation clearly states that "using internal and external pools at the same time has not been tested and may result in problematic behaviour" - in fact, this works in test env, and results in desired effect of pooling writer connections in external pool and reader connections in internal pool. However, this setup cannot be used in production as it's not recommended by the driver developers.

### Proposed Solution

* introduce a driver property to replace cluster URL with writer node URL in ReadWriteSplittingPlugin
* in ReadWriteSplittingPlugin.connectInternal change the url before calling connectFunc.call() if property is set.

### Other Information

this is the flow for a connection with cluster URL:
* https://github.com/aws/aws-advanced-jdbc-wrapper/blob/main/wrapper/src/main/java/software/amazon/jdbc/plugin/readwritesplitting/ReadWriteSplittingPlugin.java#L146ends up at https://github.com/aws/aws-advanced-jdbc-wrapper/blob/main/wrapper/src/main/java/software/amazon/jdbc/plugin/DefaultConnectionPlugin.java#L179 , where a decision is made whether to use HikariPooledConnectionProvider.
* https://github.com/aws/aws-advanced-jdbc-wrapper/blob/main/wrapper/src/main/java/software/amazon/jdbc/HikariPooledConnectionProvider.java#L156 tries to determine RdsUrlType based on the host (String) URL
* cluster URL falls into https://github.com/aws/aws-advanced-jdbc-wrapper/blob/main/wrapper/src/main/java/software/amazon/jdbc/util/RdsUtils.java#L245 which returns RDS_WRITER_CLUSTER
* https://github.com/aws/aws-advanced-jdbc-wrapper/blob/main/wrapper/src/main/java/software/amazon/jdbc/HikariPooledConnectionProvider.java#L157 rejects any types except RDS_INSTANCE.
* a Connection is created via default connection provider, not Hikari pool

*IF* before all of the above calls, ReadWriteSplittingPlugin could change the HostSpec/host from cluster URL to the writer node (which it has from topology), the writer url would result in RdsUrlType.RDS_INSTANCE and the connection would be obtained from HikariPooledConnectionProvider.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### The AWS Advanced JDBC Driver version used

2.3.6

### JDK version used

21.0.1

### Operating System and version

unix

**Comment by sergiyvamz** [comment]
Hi @ivankhoosty 

The following plugin may help to make such a replacement that you suggest. Would you consider try Aurora Initial Connection Strategy Plugin in collaboration with ReadWriteSplitting Plugin? 

https://github.com/aws/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/using-plugins/UsingTheAuroraInitialConnectionStrategyPlugin.md

Thank you!

**Comment by ivankhoosty** [comment]
Thanks for the tip. I had to remove this plugin from the list because it fails to start with requireDynamicProvider message in my local setup - we connect to aurora via elb url, which results in PgDialect and not AuroraPgDialect.... Now that I have looked at the code and deployed it, the pools are working for both writer and reader connections.

Might be helpful for ReadWriteSplittingPlugin documentation to mention its reliance on AuroraInitialConnectionStrategyPlugin.

Thanks again
